src/main/java/org/jabref/logic/crawler/StudyRepository.java

```diff
@@ -0,0 +1,344 @@
+                                                   .collect(Collectors.toList());
+        List<BibEntry> libraryEntries = parsedEntries.parallelStream()
+                                                     .filter(bibEntry -> bibEntry.getType().equals(SystematicLiteratureReviewStudyEntryType.LIBRARY_ENTRY))
+                                                     .collect(Collectors.toList());
+
+        return new Study(studyEntry, queryEntries, libraryEntries);
+    }
+
+    public Study getStudy() {
+        return study;
+    }
+
+    public void persist(List<QueryResult> crawlResults) throws IOException, GitAPIException {
+        try {
+            gitHandler.updateLocalRepository();
+        } catch (GitAPIException e) {
+            LOGGER.info("Updating repository from remote failed");
+        }
+        persistResults(crawlResults);
+        study.setLastSearchDate(LocalDate.now());
+        persistStudy();
+        try {
+            gitHandler.updateRemoteRepository("Conducted search " + LocalDate.now());
+        } catch (GitAPIException e) {
+            LOGGER.info("Updating remote repository failed");
+        }
+    }
+
+    private void persistStudy() throws IOException {
+        writeResultToFile(studyDefinitionBib, new BibDatabase(study.getAllEntries()));
+    }
+
+    /**
+     * Create for each query a folder, and for each fetcher a bib file in the query folder to store its results.
+     */
+    private void setUpRepositoryStructure() throws IOException {
+        // Cannot use stream here since IOException has to be thrown
+        LibraryEntryToFetcherConverter converter = new LibraryEntryToFetcherConverter(study.getActiveLibraryEntries(), importFormatPreferences);
+        for (String query : study.getSearchQueryStrings()) {
+            createQueryResultFolder(query);
+            converter.getActiveFetchers()
+                     .forEach(searchBasedFetcher -> createFetcherResultFile(query, searchBasedFetcher));
+            createQueryResultFile(query);
+        }
+        createStudyResultFile();
+    }
+
+    /**
+     * Creates a folder using the query and its corresponding query id.
+     * This folder name is unique for each query, as long as the query id in the study definition is unique for each query.
+     *
+     * @param query The query the folder is created for
+     */
+    private void createQueryResultFolder(String query) throws IOException {
+        Path queryResultFolder = getPathToQueryDirectory(query);
+        createFolder(queryResultFolder);
+    }
+
+    private void createFolder(Path folder) throws IOException {
+        if (Files.notExists(folder)) {
+            try {
+                Files.createDirectory(folder);
+            } catch (IOException e) {
+                throw new IOException("Error during creation of repository structure.", e);
```

Comment:
Isn't this superflous? Why not let Files.createDirectory directlry throw the excption?